### ISSUE TITLE
about language embedding

### ISSUE THREAD

**Issue by shanhaidexiamo** [bensonbs/VALL-E-X-Trainer#1]
Hi, I want to ask you about how to use language embedding  when training. As we know , in inference stage ,  language embedding  is add to text embedding , but in your forward func, it seems that you did not use language embedding? 

Thank you.

**Comment by bensonbs** [comment]
I'm sorry, I cannot answer your question, but you can ask the original author.
This repository has made some modifications to [VALL-E-X-Trainer-by-CustomData](https://github.com/0417keito/VALL-E-X-Trainer-by-CustomData)

- Fixed an error with Pathlib on Linux systems and fixed some minor bugs.
- Modified the default learning rate to reduce the likelihood of gradient vanishing
- Reduced the Maxtoken number to lower the required Vram, correspondingly, accumulate gradients of 5 batches before performing a backward pass

We fine-tuned a 50-hour single speaker Chinese dataset, and trained it for 8 hours on a Tesla V100 32GB, with the model roughly overfitting after the 4 hour. We successfully replicated the accent and voice.

However, the results generated at present are not very stable, and it takes about 5-10 attempts to obtain a satisfactory sample.

If you like this project, please support the original author.

**Comment by shanhaidexiamo** [comment]
Thank you very much for your reply! I will continue to follow the original author~

**Comment by bensonbs** [comment]
How many GB of Vram does your GPU have?

**Comment by Berthaniu** [comment]
Hi Shanhaidexiamo,

I'm also confused about how to implement the language embedding, have you gotten any response from the authors about this question?
